users/views.py

```diff
@@ -1,3 +1,36 @@
-from django.shortcuts import render
+import requests
 
-# Create your views here.
+import jwt
+
+from django.views import View
+from django.http  import JsonResponse
+
+from users.models      import User
+from faketrip.settings import SECRET_KEY, ALGORITHM
+from core.social_apis  import Kakao_Token_Error, KakaoAPI
+
+class SigninView(View):
+    def get(self, request):
+        try:
+            access_token     = request.headers.get('Authorization')
+            print(access_token)
+            user_information = KakaoAPI(access_token).get_kakao_user_information()
+            
+            kakao_pk = user_information.get('id')
+            email    = user_information.get('kakao_account').get('email')
+            nickname = user_information.get('properties').get('nickname')
+
+            if not User.objects.filter(kakao_pk = kakao_pk).exists():
+                user = User.objects.create(kakao_pk = kakao_pk, email = email, name = nickname)
+            else:
+                user = User.objects.get(kakao_pk = kakao_pk)
+                if not(user.email == email and user.name == nickname):
+                    user.email, user.name = email, nickname
+                    user.save()
```

Comment:
아래 코드로 수정하면 더 가독성이 좋아질 것 같은데 어떠신가요?
```suggestion
            user, created = User.objects.get_or_create(kakao_pk=kakao_pk, default = {"email" : email, "name" : nickname})
            
            if not created and not(user.email == email and user.name == nickname):
                user.email, user.name = email, nickname
                user.save()
```

Reply:
너무 좋은것 같습니다!